### DISCUSSION TITLE
Variable width bars

### DISCUSSION THREAD

**Discussion by pgbanfield** [Live-Charts/LiveCharts2#238] [General]
Hi,
is there a straightforward way to make a bar graph with the bar's width being dependent on the data?
I am charting events of variable duration, but I can't find a way to vary the width of bars I have tried to accomplish the same thing with a stepped line series, but then I found that I had to add additional points to the data set in order to show the areas I want to see, but then those additional points are shown in the tooltip.
Screenshot attached.

![example](https://user-images.githubusercontent.com/87652041/136996748-ff86ee6b-b844-4075-a2e5-7ae9663bddaf.PNG)
dd

**Comment by preintel** [comment]
You can possibly use sections instead of series to draw in each section:

```
cartesianChart1.Sections = new ObservableCollection<RectangularSection>
                    {
                        new RectangularSection
                        {
                            Xi = 100,
                            Xj = 120,
                            Yi = 0,
                            Yj = 4.25,
                            Fill = new LiveChartsCore.SkiaSharpView.Painting.SolidColorPaint(new SkiaSharp.SKColor(0, 255, 0))
                        },
                        new RectangularSection
                        {
                            Xi = 120,
                            Xj = 140,
                            Yi = 0,
                            Yj = 4.25,
                            Fill = new LiveChartsCore.SkiaSharpView.Painting.SolidColorPaint(new SkiaSharp.SKColor(255, 0, 0))
                        }
                    };
```

Another approach could be to modify the tooltip label itself, change the line (p => $"{p.PrimaryValue:N2}" to handle the tool tip variable p as you want, consider a method to handle the primaryvalue property and redisplay it accordingly.

```
cartesianChart.Series = new List<ISeries>
                                                     {
                                                          new LineSeries<ObservablePoint>
                                                         {
                                                              Name = "MyChart",
                                                              Values = myList<ObservablePoint>,
                                                              TooltipLabelFormatter = p => $"{p.PrimaryValue:N2}",
                                                              Stroke = new LiveChartsCore.SkiaSharpView.Painting.SolidColorPaint(new SkiaSharp.SKColor(0, 0, 255)) { StrokeThickness = 1 },
                                                              Fill = null,
                                                              GeometrySize = 1
                                                         }
                                                     };
```

**Reply by pgbanfield** [reply]
I've been experimenting with changing the tooltip text, and have managed to get it to display the text I want for the non-zero values.
My one remaining problem is that it still prints an empty line for the zero points as shown in the attached screenshot. 
Is there any way to prevent these from being displayed?

![tooltips](https://user-images.githubusercontent.com/87652041/139662446-15b0ec4b-c763-4e89-9c2a-aaae8bc5c876.PNG)
